test/Microsoft.AspNet.Mvc.FunctionalTests/TryValidateModelTest.cs

```diff
@@ -87,5 +81,37 @@ public class TryValidateModelTest
             var body = await response.Content.ReadAsStringAsync();
             Assert.Equal("{}", body);
         }
+
+        [Fact]
+        public async Task TryValidateModel_CollectionsModel_ReturnsErrorsForInvalidProperties()
```

Comment:
I updated `TryValidateModelAfterClearingValidationErrorInParameter` to call `ClearValidationState` on the key to clear the errors. Btw, this test does 2 things - clears validation errors explicity on action parameter and calls `TryValidateModel` on model.